### ISSUE TITLE
Dart2js failures for some NNBD tests

### ISSUE THREAD

**Issue by sgrekhov** [dart-lang/sdk#41502]
For the following test (this is not the complete list just a part but all failures are very similar) dart2js fails

co19/LanguageFeatures/nnbd/type-aliases/extension_method_resolution_A01_t06
co19/LanguageFeatures/nnbd/type-normalization/norm_A11_t01
co19/LanguageFeatures/nnbd/type-normalization/norm_A12_t01
co19/LanguageFeatures/nnbd/type-normalization/norm_A13_t01
co19/LanguageFeatures/nnbd/type-normalization/norm_A14_t01
co19/LanguageFeatures/nnbd/type-normalization/norm_A15_t01
```
/=========================================================================================================================================================================\
| dart2js-hostasserts-strong-linux-x64-chrome:co19/LanguageFeatures/nnbd/type-aliases/extension_method_resolution_A01_t06 is new and failed (RuntimeError, expected Pass) |
\=========================================================================================================================================================================/

--- Command "dart2js" (took 03.000428s):
DART_CONFIGURATION=ReleaseX64NNBD sdk_nnbd/bin/dart2js_developer --enable-experiment=non-nullable,nonfunction-type-aliases --enable-experiment=non-nullable,nonfunction-type-aliases --test-mode --packages=/b/s/w/ir/.packages /b/s/w/ir/tests/co19/src/LanguageFeatures/nnbd/type-aliases/extension_method_resolution_A01_t06.dart --null-safety --libraries-spec=sdk_nnbd/lib/libraries.json --platform-binaries=out/ReleaseX64NNBD/ --out=/b/s/w/ir/out/ReleaseX64NNBD/generated_compilations/dart2js/tests_co19_src_LanguageFeatures_nnbd_type-aliases_extension_method_resolution_A01_t06/extension_method_resolution_A01_t06.js

exit code:
0

stdout:
Compiled 8,145,072 characters Dart to 93,503 characters JavaScript in 3.43 seconds
Dart file /b/s/w/ir/tests/co19/src/LanguageFeatures/nnbd/type-aliases/extension_method_resolution_A01_t06.dart compiled to JavaScript: out/ReleaseX64NNBD/generated_compilations/dart2js/tests_co19_src_LanguageFeatures_nnbd_type-aliases_extension_method_resolution_A01_t06/extension_method_resolution_A01_t06.js

--- Command "chrome" (took 97ms):
/b/s/w/ir/tools/sdks/dart-sdk/bin/dart pkg/test_runner/bin/launch_browser.dart chrome http://127.0.0.1:45177/root_build/generated_tests/dart2js-chrome/tests_co19_src_LanguageFeatures_nnbd_type-aliases_extension_method_resolution_A01_t06/test.html?crossOriginPort=44125

Runtime window.onerror:
window.onerror called:

http://127.0.0.1:45177/root_build/generated_compilations/dart2js/tests_co19_src_LanguageFeatures_nnbd_type-aliases_extension_method_resolution_A01_t06/extension_method_resolution_A01_t06.js:1117:
Uncaught TypeError: Cannot read property 'Object' of undefined



Deobfuscated error and stack:
<no error message found>
    at window.onerror called:                                         unparsed
    at <fn>                                                           out/ReleaseX64NNBD/generated_compilations/dart2js/tests_co19_src_LanguageFeatures_nnbd_type-aliases_extension_method_resolution_A01_t06/extension_method_resolution_A01_t06.js 1117
    at Uncaught TypeError: Cannot read property 'Object' of undefined unparsed


exit code:
0

--- To debug locally, run:
/b/s/w/ir/tools/sdks/dart-sdk/bin/dart /b/s/w/ir/pkg/test_runner/bin/http_server.dart -p 45177 -c 44125 --network 127.0.0.1 --build-directory=/b/s/w/ir/out/ReleaseX64NNBD/ --runtime=chrome --packages=/b/s/w/ir/.packages

--- Re-run this test:
python tools/test.py -n dart2js-hostasserts-strong-linux-x64-chrome --dart2js-batch co19/LanguageFeatures/nnbd/type-aliases/extension_method_resolution_A01_t06

/========================================================================================================================================================\
| dart2js-hostasserts-strong-linux-x64-chrome:co19/LanguageFeatures/nnbd/type-normalization/norm_A11_t01 is new and failed (RuntimeError, expected Pass) |
\========================================================================================================================================================/

--- Command "dart2js" (took 03.000695s):
DART_CONFIGURATION=ReleaseX64NNBD sdk_nnbd/bin/dart2js_developer --enable-experiment=non-nullable --enable-experiment=non-nullable --test-mode --packages=/b/s/w/ir/.packages /b/s/w/ir/tests/co19/src/LanguageFeatures/nnbd/type-normalization/norm_A11_t01.dart --null-safety --libraries-spec=sdk_nnbd/lib/libraries.json --platform-binaries=out/ReleaseX64NNBD/ --out=/b/s/w/ir/out/ReleaseX64NNBD/generated_compilations/dart2js/tests_co19_src_LanguageFeatures_nnbd_type-normalization_norm_A11_t01/norm_A11_t01.js

exit code:
0

stdout:
Hint: When run on the command-line, the compiled output might require a preamble file located in:
  <sdk>/lib/_internal/js_runtime/lib/preambles.
Compiled 8,091,930 characters Dart to 144,693 characters JavaScript in 3.69 seconds
Dart file /b/s/w/ir/tests/co19/src/LanguageFeatures/nnbd/type-normalization/norm_A11_t01.dart compiled to JavaScript: out/ReleaseX64NNBD/generated_compilations/dart2js/tests_co19_src_LanguageFeatures_nnbd_type-normalization_norm_A11_t01/norm_A11_t01.js

--- Command "chrome" (took 93ms):
/b/s/w/ir/tools/sdks/dart-sdk/bin/dart pkg/test_runner/bin/launch_browser.dart chrome http://127.0.0.1:35223/root_build/generated_tests/dart2js-chrome/tests_co19_src_LanguageFeatures_nnbd_type-normalization_norm_A11_t01/test.html?crossOriginPort=40425

Runtime window.onerror:
window.onerror called:

http://127.0.0.1:35223/root_build/generated_compilations/dart2js/tests_co19_src_LanguageFeatures_nnbd_type-normalization_norm_A11_t01/norm_A11_t01.js:1362:
Uncaught TypeError: Cannot read property 'Object' of undefined



Deobfuscated error and stack:
<no error message found>
    at window.onerror called:                                         unparsed
    at <fn>                                                           out/ReleaseX64NNBD/generated_compilations/dart2js/tests_co19_src_LanguageFeatures_nnbd_type-normalization_norm_A11_t01/norm_A11_t01.js 1362
    at Uncaught TypeError: Cannot read property 'Object' of undefined unparsed


exit code:
0

--- To debug locally, run:
/b/s/w/ir/tools/sdks/dart-sdk/bin/dart /b/s/w/ir/pkg/test_runner/bin/http_server.dart -p 35223 -c 40425 --network 127.0.0.1 --build-directory=/b/s/w/ir/out/ReleaseX64NNBD/ --runtime=chrome --packages=/b/s/w/ir/.packages

--- Re-run this test:
python tools/test.py -n dart2js-hostasserts-strong-linux-x64-chrome --dart2js-batch co19/LanguageFeatures/nnbd/type-normalization/norm_A11_t01

```

**Comment by vsmenon** [comment]
fyi - @fishythefish @rakudrama

**Comment by fishythefish** [comment]
These are NNBD strong mode failures, so this is currently expected. FWIW https://dart-review.googlesource.com/c/sdk/+/143360 will fix ~8000 of these.

**Comment by fishythefish** [comment]
Should be resolved by 1ac89cd5b3ed6dde5830fc3c3b8c14a07ee79916. Feel free to (re)open an issue if there are more specific failures I should look into, but the work on strong mode is still ongoing and we don't expect all the tests to be passing while we're still burning down issues.
